packages/web-api-scan-runner/src/crawler/discovered-urls-processor.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { ServiceConfiguration } from 'common';
+import { inject, injectable } from 'inversify';
+import _, { isNil } from 'lodash';
+import { WebsiteScanResult } from 'storage-documents';
+
+@injectable()
+export class DiscoveredUrlsProcessor {
+    public constructor(@inject(ServiceConfiguration) private readonly serviceConfig: ServiceConfiguration) {}
+
+    public async getProcessedUrls(discoveredUrls: string[], websiteScanResult?: WebsiteScanResult): Promise<string[]> {
+        let processedUrls = discoveredUrls;
+        if (isNil(discoveredUrls)) {
+            return [];
+        }
+
+        if (!isNil(websiteScanResult) && !isNil(websiteScanResult.knownPages)) {
+            processedUrls = this.removeUrlsFromList(discoveredUrls, websiteScanResult.knownPages);
+        }
+        processedUrls = this.limitNumUrls(processedUrls, await this.getUrlLimit(websiteScanResult));
```

Comment:
I'm not wild about the complexity of the line here. But more importantly, we should only ever need to call getUrlLimit once. We should probably pass it into the class as a parameter for example.